<launch>

    <!--args for mavros tf-->
    <arg name="quad_tf" default="/uav0_wrt_world"/>
    <arg name="world_tf" default="/world_enu"/>
    <arg name="offset_x" default="0"/>
    <arg name="offset_y" default="0"/>

    <!--args for apriltag_position_pub.py-->
    <arg name="uav_name" default="uav0"/>
    <arg name ="camera_name" default="camera_name"/>
    <arg name="id_tag" default="0"/>
    <arg name="rtagdrone" default="/tag_wrt_uav0"/>
    <arg name="frame_tag_id" default="/tag_0"/>
    <arg name="height_from_tag" default="1"/>

    <arg name="old_target_tf" default="downwards_custom_0_optical"/>

    <node pkg="utm" type="drone_transformation.py" name="drone_transformation">
        <param name="quad_tf" value="$(arg quad_tf)"/>
        <param name="old_target_tf" value="$(arg old_target_tf)"/>
        <param name="world_tf" value="$(arg world_tf)"/>
    </node>

    <!--comment this out if you don't want apriltag tag detection-->
    <remap from="/tag_detections" to="/$(arg uav_name)/tag_detections"/>
    <include file="$(find apriltag_ros)/launch/continuous_detection.launch">
        <arg name="camera_name" value="$(arg camera_name)"/>
    </include>

    <!--comment this out if you don't want apriltag tag detection-->
    <node pkg="utm" type="apriltag_position_pub.py" name="apriltag_position_pub">
        <param name="tag_id" value="$(arg id_tag)"/>
        <param name="rtag_drone" value="$(arg rtagdrone)"/>
        <param name="quad_tf" value="$(arg quad_tf)"/>
        <param name="tag_frame_id" value="$(arg frame_tag_id)"/> 
        <param name="alt_from_tag" value="$(arg height_from_tag)"/>    
    </node>

    <!--comment this out if you don't want apriltag tag detection-->
    <node pkg="utm" type="ekf_tag.py" name="ekf_tag"/>

</launch>